### ISSUE TITLE
Expose --logging flag for minshift start command

### ISSUE THREAD

**Issue by praveenkumar** [minishift/minishift#811]
Currently we don't expose `logging` as part of minishift start but `oc cluster up` provide this functionality as experimental same as `metrics`. We should also enable it to give user a chance to play around with it.

```
$ oc cluster up -h
[...]
      --logging=false: Install logging (experimental)
      --metrics=false: Install metrics (experimental)
      --no-proxy=[]: List of hosts or subnets for which a proxy should not be used
[...]
```

**Comment by hferentschik** [comment]
Testing the pull request, I see the same issue as with Metrics. 'cluster up' completes, but the aggregated logging does not seem to work:

    $ minishift start --logging
    Starting local OpenShift cluster using 'virtualbox' hypervisor...
    ....
    -- Installing logging ... OK
    ....
    The kibana logging UI is available at:
       https://kibana-logging.192.168.99.109.nip.io

When I try to access Kibana, I get:

    Application is not available

    The application is currently not serving requests at this endpoint. It may not have been started or is still starting.

It seems there are several logging related pods running though:

    ec02477620e9        openshift/origin-logging-curator:v1.5.0         "sh run.sh"              17 minutes ago      Up 17 minutes                           k8s_curator.b05acae2_logging-curator-1-t24rw_logging_295bcf9b-2ff7-11e7-bd55-d211ede371ff_01b74627
    06f2b6b90e21        openshift/origin-logging-elasticsearch:v1.5.0   "sh /opt/app-root/src"   18 minutes ago      Up 18 minutes                           k8s_elasticsearch.3f5b3d4f_logging-es-4nyoayni-1-gtnr5_logging_2998450f-2ff7-11e7-bd55-d211ede371ff_93a9a413
    ba4caacb3be8        openshift/origin-logging-fluentd:v1.5.0         "sh run.sh"              20 minutes ago      Up 20 minutes                           k8s_fluentd-elasticsearch.d69f20a1_logging-fluentd-   r7z74_logging_27b028e5-2ff7-11e7-bd55-d211ede371ff_501bc71e
    f9e67751752b        openshift/origin-pod:v1.5.0                     "/pod"                   21 minutes ago      Up 21 minutes                           k8s_POD.49de27a2_logging-es-4nyoayni-1-gtnr5_logging_2998450f-2ff7-11e7-bd55-d211ede371ff_dd2bc295
    a5f92d123981        openshift/origin-pod:v1.5.0                     "/pod"                   21 minutes ago      Up 21 minutes                           k8s_POD.4ebadca1_logging-curator-1-t24rw_logging_295bcf9b-2ff7- 11e7-bd55-d211ede371ff_505af755
    ad48b2479455        openshift/origin-pod:v1.5.0                     "/pod"                   21 minutes ago      Up 21 minutes                           k8s_POD.4ebadca1_logging-fluentd-r7z74_logging_27b028e5-2ff7-11e7-bd55-d211ede371ff_3f05b3c3

This looks very similar to the deployment problem for metrics.

**Comment by hferentschik** [comment]
See also issue #802 regarding metrics not starting properly.

**Comment by hferentschik** [comment]
@csrwng, is this basically the same issue as with #802? If so, I would think that this are issues with the metrics resp. logging deployer. Where would one report this?

**Comment by csrwng** [comment]
@hferentschik I haven't tried it recently but it sounds like it could be.

**Comment by hferentschik** [comment]
> I haven't tried it recently but it sounds like it could be.

:-( I would like to understand what's going on here? AFAIK there is some sort of deployer pod which in turn will make sure that other pods get installed and wired together. Is this correct? So why can this deployer pod not make sure that things happen in the right sequence? Or have some other sort of error handling and detection?  Success of deployment should not be dependent on network latency and how fast or slow certain pods can start. 

Given that Metrics and Aggregated Logging are important features of the OpenShift value, I really think this needs to be addressed at the right place. Wherever that is.

**Comment by csrwng** [comment]
Yes, I think that the deployer script should allow for the image pull to take a long time. Maybe with exponential backoff, instead of just giving up after a set period of time. If needed, I'll work on a PR for the appropriate repositories.

**Comment by hferentschik** [comment]
>  I think that the deployer script should allow for the image pull to take a long time. Maybe with exponential backoff, instead of just giving up after a set period of time.

+1

>  If needed, I'll work on a PR for the appropriate repositories.

Sounds awesome. Please let us know what these "appropriate repositories" are. Maybe we can lend a hand as well or at least provide some input.

**Comment by csrwng** [comment]
https://github.com/openshift/origin-metrics
https://github.com/openshift/origin-aggregated-logging

**Comment by hferentschik** [comment]
Thanks :-)

**Comment by hferentschik** [comment]
Merged via pull request #853
